firmware/applications/SignBuddy/main.c

```diff
@@ -64,17 +65,26 @@ static void board_bringup(void)
   ble_uart_init();
 }
 
+static void led_process(void)
+{
+  static uint32_t last_ticks = 0;
+  static uint32_t led_state  = 0;
+
+  uint32_t time = system_time_get();
+
+  if (system_time_cmp_ms(last_ticks, time) < 1000) {
+    led_state = (led_state + 1) % 2;
+    gpio_led_set(led_state);
```

Comment:
Not sure about this logic, I think it would make more sense to do something like this:
`  if (system_time_cmp_ms(last_ticks, time) > 1000) {`
`    led_state = !led_state;`
`    gpio_led_set(led_state);`
`    last_ticks = time;`
`  }`

Reply:
yeah, you're right that should've been the exit condition

```
if (system_time_cmp_ms(last_ticks, time) > 1000) {
  return;
}
...
```